core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilder.java

```diff
@@ -522,4 +525,33 @@ public String toString() {
             return name().toLowerCase(Locale.ROOT);
         }
     }
+
+    public static enum BoundaryScannerType implements Writeable {
+        SIMPLE, BREAK_ITERATOR;
```

Comment:
Got it thanks. Though we could avoid creating another option and merge the logic in a single one ? The boundary scanner type could be any of:
* simple
* sentence
* word
The fact that we use a break iterator for sentence and word is just an implementation detail ? 
Although "CHARACTER" seems weird, should we really expose it ?